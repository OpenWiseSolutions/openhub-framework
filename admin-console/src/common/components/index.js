export { default as LoginCard } from './LoginCard/LoginCard'
