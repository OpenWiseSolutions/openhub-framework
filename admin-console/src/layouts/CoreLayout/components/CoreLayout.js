import React, { PropTypes, Component } from 'react'
import Radium, { StyleRoot } from 'radium'
import Navbar from '../../../common/components/Navbar/Navbar'
import Sidebar from '../../../common/components/Sidebar/Sidebar'
import styles from './coreLayout.styles'
import LoginModal from '../../../common/containers/loginModal.container'

@Radium
class CoreLayout extends Component {

  render () {
    const {
      authUser,
      children,
      sidebarExtended,
      navbarUserExpanded,
      actions,
      authActions
    } = this.props

    const bodyStyles = [
      styles.body,
      sidebarExtended && authUser && styles.body.extended
    ]

    return (
      <StyleRoot>
        <div style={styles.main}>
          <Sidebar extended={sidebarExtended && !!authUser} />
          <LoginModal />
          <div style={bodyStyles}>
            <Navbar
              authUser={authUser}
              logout={authActions.logout}
              navbarUserExpanded={navbarUserExpanded}
              toggleUser={actions.toggleNavbarUser}
              toggleLoginModal={authActions.toggleLoginModal}
              toggleSidebar={actions.toggleSidebar} />
            <div style={styles.contents} >
              {children}
            </div>
          </div>
        </div>
      </StyleRoot>
    )
  }
}

CoreLayout.propTypes = {
  children: PropTypes.element.isRequired,
  sidebarExtended: PropTypes.bool,
  actions: PropTypes.object,
  authActions: PropTypes.object,
  navbarUserExpanded: PropTypes.bool,
  authUser: PropTypes.object
}

export default CoreLayout
