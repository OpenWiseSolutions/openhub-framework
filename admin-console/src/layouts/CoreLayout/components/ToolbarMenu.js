import React, { Component } from 'react'
import PropTypes from 'prop-types'
import List from 'react-md/lib/Lists/List'
import ListItem from 'react-md/lib/Lists/ListItem'
import MenuButton from 'react-md/lib/Menus/MenuButton'
import Menu from 'react-md/lib/Menus/Menu'

class ToolbarMenu extends Component {
  render () {
    const { userData, logoutUser } = this.props
    if (!userData) return null
    const { fullName } = userData
    return (
      <div style={{ height: '100%', display: 'flex', justifyContent: 'center', flexDirection: 'column' }} >
        <MenuButton flat position={Menu.Positions.BELOW} label={fullName} id='static-1' >
          <List>
            <ListItem onClick={logoutUser} primaryText={'Logout'} />
          </List >
        </MenuButton >
      </div >
    )
  }
}

ToolbarMenu.propTypes = {
  userData: PropTypes.object,
  logoutUser: PropTypes.func
}

export default ToolbarMenu
