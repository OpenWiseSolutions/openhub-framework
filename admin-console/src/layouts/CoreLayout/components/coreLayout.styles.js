import { sidebarWidth } from '../../../common/components/Sidebar/sidebar.styles'
import { transition } from '../../../styles/constants'

export default {
  main: {
    position: 'absolute',
    height: '100%',
    width: '100%'
  },
  body: {
    ...transition(),
    position: 'absolute',
    width: '100%',
    height: '100%',
    boxSizing: 'border-box',
    overflow: 'auto',
    extended: {
      paddingLeft: `${sidebarWidth}px`
    }
  }
}

