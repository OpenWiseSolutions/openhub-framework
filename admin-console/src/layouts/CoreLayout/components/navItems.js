import React from 'react'
import { hashHistory } from 'react-router'
import { path } from 'ramda'
import ListItem from 'react-md/lib/Lists/ListItem'
import FontIcon from 'react-md/lib/FontIcons'

const isActive = (url) => url === window.location.hash.substr(1)

export default (config) => {
  if (!config) return null
  return [
    <ListItem
      key='Dashboard'
      active={isActive('/')}
      onClick={() => hashHistory.push('/')}
      leftIcon={<FontIcon >home</FontIcon >}
      primaryText={'Dashboard'}
    />,

    // Analytics
    <div key='Analytics' >
      {config.analytics &&
      <ListItem
        primaryText={'Analytics'}
        defaultOpen
        leftIcon={<FontIcon >blur_linear</FontIcon >}
        nestedItems={[
          path(['analytics', 'messages', 'enabled'], config) && <ListItem
            key='Messages'
            active={isActive('/messages')}
            onClick={() => hashHistory.push('/messages')}
            leftIcon={<FontIcon >drafts</FontIcon >}
            primaryText={'Messages'}
          />
        ]}
      />}
    </div >,

    // Infrastructure
    <div key='Infrastructure'>
      {config.cluster &&
      <ListItem
        primaryText={'Infrastructure'}
        leftIcon={<FontIcon >developer_board</FontIcon >}
        defaultOpen
        nestedItems={[
          path(['cluster', 'nodes', 'enabled'], config) && <ListItem
            key='Nodes'
            active={isActive('/nodes')}
            onClick={() => hashHistory.push('/nodes')}
            leftIcon={<FontIcon >device_hub</FontIcon >}
            primaryText={'Nodes'}
          />,
          path(['infrastructure', 'services', 'wsdl', 'enabled'], config) && <ListItem
            key='WSDL'
            active={isActive('/wsdl')}
            onClick={() => hashHistory.push('/wsdl')}
            leftIcon={<FontIcon >device_hub</FontIcon >}
            primaryText={'WSDL'}
          />
        ]}
      />}
    </div >,

    // Configuration
    <ListItem
      key='Configuration'
      primaryText={'Configuration'}
      leftIcon={<FontIcon >build</FontIcon >}
      defaultOpen
      nestedItems={[
        path(['configuration', 'systemParams', 'enabled'], config) && <ListItem
          key='params'
          active={isActive('/config-params')}
          onClick={() => hashHistory.push('/config-params')}
          leftIcon={<FontIcon >extension</FontIcon >}
          primaryText={'Config Params'}
        />,
        path(['configuration', 'logging', 'enabled'], config) && <ListItem
          key='logging'
          onClick={() => hashHistory.push('/config-logging')}
          active={isActive('/config-logging')}
          leftIcon={<FontIcon >info</FontIcon >}
          primaryText={'Config Logging'}
        />,
        path(['configuration', 'environment', 'enabled'], config) && <ListItem
          key='properties'
          onClick={() => hashHistory.push('/environment-properties')}
          active={isActive('/environment-properties')}
          leftIcon={<FontIcon >filter_hdr</FontIcon >}
          primaryText={'Environment Properties'}
        />,
        path(['configuration', 'alerts', 'enabled'], config) && <ListItem
          key='alerts'
          onClick={() => hashHistory.push('/alerts')}
          active={isActive('/alerts')}
          leftIcon={<FontIcon >notifications_active</FontIcon >}
          primaryText={'Alerts'}
        />,
        path(['configuration', 'errorCodeCatalog', 'enabled'], config) && <ListItem
          key='errors'
          onClick={() => hashHistory.push('/errors-overview')}
          active={isActive('/errors-overview')}
          leftIcon={<FontIcon >bug_report</FontIcon >}
          primaryText={'Errors Overview'}
        />
      ]}
    />,

    { divider: true },

    // External
    <div key='External' >
      {path(['externalLinks', 'enabled'], config) &&
      <ListItem
        primaryText={'External'}
        leftIcon={<FontIcon >link</FontIcon >}
        nestedItems={
          config.externalLinks.items.map(({ title, link, enabled }) => {
            if (!enabled) return null
            return (
              <ListItem
                key={title}
                onClick={() => window.open(link, '_blank')}
                leftIcon={<FontIcon >keyboard_arrow_right</FontIcon >}
                primaryText={title}
              />)
          })
        }
      />}
    </div >,

    // Changes
    <div key='Changes' >
      {path(['changes', 'enabled'], config) && <ListItem
        active={isActive('/changes')}
        onClick={() => hashHistory.push('/changes')}
        leftIcon={<FontIcon >merge_type</FontIcon >}
        primaryText={'Changes'}
      />}
    </div >

  ]
}
