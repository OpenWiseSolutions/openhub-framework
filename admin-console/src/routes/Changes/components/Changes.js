import React, { PropTypes, Component } from 'react'
import Radium from 'radium'
import ReactMarkDown from 'react-markdown'
import Panel from '../../../common/components/Panel/Panel'
import styles from './changes.styles.js'

@Radium
class Changes extends Component {

  componentDidMount () {
    const { getChanges } = this.props
    getChanges()
  }

  render () {
    const { changesData } = this.props

    return (
      <Panel style={styles.main} title={'Changes'}>
        <div style={styles.markdown}>
          {changesData && <ReactMarkDown source={changesData} />}
        </div>
      </Panel>
    )
  }
}

Changes.propTypes = {
  getChanges: PropTypes.func,
  changesData: PropTypes.string
}

export default Changes
