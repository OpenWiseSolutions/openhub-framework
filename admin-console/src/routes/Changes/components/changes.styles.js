import { gap } from '../../../styles/constants'

export default {
  main:{
    width: '98%'
  },
  markdown: {
    paddingLeft: gap,
    paddingRight: gap,
    paddingTop: gap
  }
}
