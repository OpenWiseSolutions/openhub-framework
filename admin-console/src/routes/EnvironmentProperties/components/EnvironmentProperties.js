import React, { PropTypes, Component } from 'react'
import Radium from 'radium'
import styles from './environmentProperties.styles.js'
import ReactJson from 'react-json-view'
import Panel from '../../../common/components/Panel/Panel'

@Radium
class EnvironmentProperties extends Component {
  componentDidMount () {
    const { getEnvironmentData, getConfigData } = this.props
    getConfigData()
    getEnvironmentData()
  }

  render () {
    const {
      environmentData, configData
    } = this.props

    return (
      <div style={styles.main}>
        <Panel style={styles.panel} title={'Environment properties'}>
          <div style={styles.json}>
            {environmentData && <ReactJson src={environmentData} />}
          </div>
        </Panel>
        <Panel style={styles.panel} title={'Config properties'}>
          <div style={styles.json}>
            {configData && <ReactJson src={configData} />}
          </div>
        </Panel>
      </div>
    )
  }
}

EnvironmentProperties.propTypes = {
  getEnvironmentData: PropTypes.func,
  getConfigData: PropTypes.func,
  environmentData: PropTypes.object,
  configData: PropTypes.object
}

export default EnvironmentProperties
