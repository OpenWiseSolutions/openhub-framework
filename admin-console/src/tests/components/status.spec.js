import React from 'react'
import renderer from 'react-test-renderer'
import { Status } from '../../common/components'

describe('Status Component', () => {
  test('renders status UP', () => {
    const tree = renderer.create(
      <Status status />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('renders status DOWN', () => {
    const tree = renderer.create(
      <Status status={false} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
