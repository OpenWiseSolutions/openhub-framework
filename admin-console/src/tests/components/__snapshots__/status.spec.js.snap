// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status Component renders status DOWN 1`] = `
<div
  data-radium={true}
  style={
    Object {
      "fontWeight": 800,
    }
  }
>
  <span
    data-radium={true}
    style={
      Object {
        "color": "#F04A44",
      }
    }
  >
     DOWN 
  </span>
</div>
`;

exports[`Status Component renders status UP 1`] = `
<div
  data-radium={true}
  style={
    Object {
      "fontWeight": 800,
    }
  }
>
  <span
    data-radium={true}
    style={
      Object {
        "color": "#5fac40",
      }
    }
  >
     UP 
  </span>
</div>
`;
